### PR TITLE
azure-cloud-provider: skip test if only config or docs are updated

### PR DESCRIPTION
**Pull request by MartinForReal**
skip test if only config or docs are updated

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@ presubmits:
   kubernetes-sigs/cloud-provider-azure:
   - name: pull-cloud-provider-azure-check
     always_run: true
+    skip_if_only_changed: "^docs/|^site/|^\\.github/|\\.(md|adoc)$|^(README|LICENSE)$"
```
Comment (andyzhangx) [inline config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml:5]: will this work? I would also like to configure this for the other upstream repos if it works.

File: config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml [hunk 2]
```diff
@@ -260,6 +265,7 @@ presubmits:
       testgrid-num-columns-recent: '30'
   - name: pull-cloud-provider-azure-e2e-capz
     always_run: false
+    #skip_if_only_changed: "^docs/|^site/|^\\.github/|\\.(md|adoc)$|^(README|LICENSE)$"
```
Comment (andyzhangx) [inline config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml:268]: commented?

Comment (MartinForReal) [reply config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml:268]: always_run: false and skip_if_only_changed are mutually exclusive

File: config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml (hunk 1 above)
Comment (MartinForReal) [reply config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml:5]: Please refer to https://github.com/kubernetes/test-infra/blob/c341b223083a7e5766be99620eba58cc3c8142f1/prow/jobs.md#triggering-jobs-based-on-changes
I have no idea how to test it.

**Review by andyzhangx — APPROVED** [review]
/ok-to-test
/lgtm
/approve

File: config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml (hunk 1 above)
Comment (andyzhangx) [reply config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml:5]: sure so let's merge it first

Comment (MartinForReal) [reply config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml:4]: I think it works. Please refer to https://github.com/kubernetes-sigs/cloud-provider-azure/pull/1139

**Pull request merged**